packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -140,6 +141,23 @@ export const Product = ({
     [setCreateTokenError]
   );
 
+  // clear any error rendered with `ErrorMessage` on form change
+  const onChangeErrorDismiss = useCallback(() => {
+    if (createTokenError.error) {
+      setCreateTokenError({ type: '', error: false });
+    } else if (
+      createSubscriptionStatus.error !== null &&
+      createSubscriptionStatus.error.code === 'card_declined'
+    ) {
+      resetCreateSubscriptionError();
+    }
+  }, [
+    createTokenError,
+    createSubscriptionStatus,
+    setCreateTokenError,
+    resetCreateSubscriptionError,
+  ]);
```

Comment:
@lmorchard I originally had this callback like so:

```
  const onChangeErrorDismiss = useCallback(() => {
    setCreateTokenError({ type: '', error: false });
    resetCreateSubscriptionError();
  }, [setCreateTokenError, resetCreateSubscriptionError]);
```

... but it fired on every `onChange` event from the form. This is probably a micro-optimization since React is really fast - you think the change I made is worth keeping? It only resets the error when it needs to now, but it's kind of less scalable.

(PR is ready for review)

Reply:
Y'know, without benchmarking, I couldn't say which is more worth doing... though I'd say since you already did the work of making this callback more discriminating about what it calls, I'd just leave it this way for now?